### ISSUE TITLE
[Feature] Download speed limiting

### ISSUE THREAD

**Issue by object-Object** [gorilla-devs/GDLauncher#694]
**Is your feature request related to a problem? Please describe.**
I live in a rural area and have limited download speeds (~10 Mbps). This means that one application can easily tie up the entire bandwidth if its download speed is not / cannot be limited. Additionally, other people use the same Internet connection. Downloading a large modpack can easily use up the entire bandwidth for an extended period of time, essentially preventing others from accessing the Internet until it's finished.

**Describe the solution you'd like.**
I'd like a setting to be added to allow limiting download speeds by setting a maximum speed (example: `800` KB/s). Steam has a good implementation of this, for example. A drop-down with predetermined values would work, or a text field where any value may be entered.

**Comment by Eskaan** [comment]
Isnt the concurrent downloads setting enough of a limit?

**Comment by object-Object** [comment]
I don't think so. Even if it's set to one download at a time, that one download can still use all of the available bandwidth.
